core/src/main/java/io/grpc/internal/ManagedChannelOrphanWrapper.java

```diff
@@ -29,6 +29,13 @@
 import java.util.logging.LogRecord;
 import java.util.logging.Logger;
 
+/**
+ *  Best effort detecting channels that has not been properly cleaned up.
+ *  Use {@link WeakReference} to avoid keeping the channel alive and retaining too much memory.
+ *  Check lost references only on new channel creation or JVM shutdown and log message to indicate
```

Comment:
I don't think we check anything during JVM shutdown. So we should probably remove the text about that part.